File: Analysis of algorithms Exam/selection.java
import java.util.*;
public class selection {
    static void readArray(int a[],int n){
        Scanner sc=new Scanner(System.in);
        for(int i=0;i<n;i++){
            a[i]=sc.nextInt();
        }
    }
        static void displayArray(int a[],int n){
       System.out.println("Array details");
       System.out.println();
       System.out.print("{"+"\t");
        for(int i=0;i<n;i++){
            System.out.print(a[i]+"\t");
        }
        System.out.print("}");
        System.out.println();
    }
    static void SelectionSort(int a[],int n){
        for(int i=0;i<n;i++){
            int j=i;
            for(int k=i+1;k<n;k++ ){
                if(a[j]>a[k]){
                    j=k;
                }
            }
            int t=a[i];
            a[i]=a[j];
            a[j]=t;
            System.out.println("__________________________________________________________________________________");
            System.out.println();
            System.out.println("after pass "+(i+1));
            displayArray(a, n);
            System.out.println();
        }
    }
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        System.out.println("enter the size of the array:");
        int n=sc.nextInt();
        int a[]=new int[n];
        System.out.println("__________________________________________________________________________________");
        System.out.println("enter the array details");
        readArray(a, n);
        System.out.println("__________________________________________________________________________________");
        displayArray(a, n);
        System.out.println("__________________________________________________________________________________");
        SelectionSort(a, n);
        System.out.println("__________________________________________________________________________________");
        displayArray(a, n);
    }
    
}

File: selection.c
#include<stdio.h>
#include<conio.h>
void readArray(int n,int array[n])
{
    printf("\nEnter the array elements :\n");
    for(int i=0;i<n;i++)
    {
     scanf("%d",&array[i]);
    }
    
}
void displayArray(int n,int array[n])
{
    printf("\nThe array elements are as follows:\n");
    for(int i=0;i<n;i++)
    {
    printf("%d \n",array[i]); 
    }
    
}
void selectionSort(int n, int array[n])
{
for(int i=0;i<n;i++)
{
    int j=i;
    for(int k=i+1;k<n;k++)
    {
        if(array[k]<array[j])
        {
            j=k;
        }
      
    }
        printf("after iteration %d",i+1);
        int temp=array[i];
        array[i]=array[j];
        array[j]=temp;
        displayArray(n,array);
}
}
void main(){
   int n;
   printf("enter the array size:");
   scanf("%d",&n);
   int array[n];
   readArray(n,array);
   printf("\nbefore swapping the array elements:\n");
   displayArray(n,array);
   selectionSort(n,array);
   displayArray(n,array);
}